### PR TITLE
delete and deprecate string and tab package in qio

### PR DESCRIPTION
**Pull request by ChristinaXu2017**
# Description

- deleted org.qcmg.string from qio, which is replaced by org.qcmg.qio.record created on #209.
- deprecate org.qcmg.tab in qio, which is replaced by org.qcmg.qio.record.
- deprecated some qmule classes which using old org.qcmg.tab.  Here qmule contains a list of dated classes and may delete on the next pull requests.
- updated q3tiledaligner due to the above two issues.
- updated qmaftools due to the above two issues.
- updated qsv due to the above two issues. 

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Test A: existing unit tests already covered these IO issues
- [X] Test B: run q3tiledalginer on reference file and then compare to adama nightly

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: qmaftools/src/org/qcmg/maf/MafFilter.java [hunk 1]
```diff
@@ -148,20 +137,19 @@ private void loadKRASFile(String krasFile) throws Exception {
...
 							
 							if (chr.equals(chr2) && position.equals(position2) && id.equals(id2)) {
 								// remove from list
 								recordAlreadyInLowerConfList = true;
-//								logger.info("removing record from low conf file: " + probableNoiseMafs.remove(tr));
 								logger.info("moving record from low conf to high conf, and updating verification status to Valid: " 
 										+ probableNoiseMafs.remove(tr));
 								
-								tr.setData(tr.getData().replaceAll("Unknown", "Valid"));
+								tr = tr.replaceAll("Unknown", "Valid");
```
Comment (holmeso) [inline qmaftools/src/org/qcmg/maf/MafFilter.java:152]: This looks suspicious

File: qmaftools/src/org/qcmg/maf/MafFinalFilter.java [hunk 2]
```diff
@@ -76,24 +68,19 @@ private void filterMafFile(String inputFile, String outputMafFile) throws Except
 						if (validationStatus.startsWith("False")) continue;
 						if (validationStatus.startsWith("Unknown")) {
 							// don't want the extra info in the validation status field - just Unknown 
-							rec.setData(rec.getData().replace(validationStatus, "Unknown"));
+							rec = rec.replace(validationStatus, "Unknown");
```
Comment (holmeso) [inline qmaftools/src/org/qcmg/maf/MafFinalFilter.java:71]: this also looks suspicious

File: qmaftools/src/org/qcmg/maf/util/MafUtils.java [hunk 3]
```diff
@@ -445,13 +428,13 @@ public static void loadDCCFile(String dccqFile, Map<ChrPosition, TorrentVerifica
 	}
 	
 	
-	public static void convertDccToMaf(TabbedRecord tabbedRecord, String patientId, String controlSampleID, 
+	public static void convertDccToMaf(String String, String patientId, String controlSampleID,
```
Comment (holmeso) [inline qmaftools/src/org/qcmg/maf/util/MafUtils.java:431]: surely we can't have a variable called `String`!

Comment (ChristinaXu2017) [reply qmaftools/src/org/qcmg/maf/util/MafUtils.java]: yes, it was typo, fixed

File: qmaftools/src/org/qcmg/maf/MafFinalFilter.java (hunk 2 above)
Comment (ChristinaXu2017) [reply qmaftools/src/org/qcmg/maf/MafFinalFilter.java]: sure, set rec to final and code changed

File: qmaftools/src/org/qcmg/maf/MafFilter.java (hunk 1 above)
Comment (ChristinaXu2017) [reply qmaftools/src/org/qcmg/maf/MafFilter.java]: sure, set rec to final and code changed

File: qmaftools/src/org/qcmg/maf/MafFilter.java [hunk 4]
```diff
@@ -138,7 +127,7 @@ private void loadKRASFile(String krasFile) throws Exception {
 								logger.info("verification DOES NOT match! - updating");
 								
 								// update record with "Valid" validation status
-								tr.setData(tr.getData().replaceAll("Unknown", "Valid"));
+								tr = tr.replaceAll("Unknown", "Valid");
```
Comment (holmeso) [inline qmaftools/src/org/qcmg/maf/MafFilter.java:130]: this is not right - you are reassigning the pointer to a new object, the collection storing the strings is not updated

Comment (ChristinaXu2017) [reply qmaftools/src/org/qcmg/maf/MafFilter.java]: yes, it was a mistake and fixed. Thanks

**Review by holmeso — APPROVED** [review]
ok

**Pull request merged**